### PR TITLE
rust: add failure() and success() helpers to ExecutionResult

### PR DESCRIPTION
**Pull request by axic**
This handles the two common cases for VMs.

### CODE REVIEW

File: bindings/rust/evmc-vm/src/lib.rs [hunk 1]
```diff
@@ -39,6 +39,14 @@ impl ExecutionResult {
         }
     }
 
+    pub fn failure() -> Self {
```
Comment (jakelang) [inline bindings/rust/evmc-vm/src/lib.rs]: Can you document these?

Comment (axic) [reply bindings/rust/evmc-vm/src/lib.rs]: Can you please?

Comment (axic) [reply bindings/rust/evmc-vm/src/lib.rs]: There's a lot to document. I suggest we create a bigger PR which documents the entire file.

Comment (axic) [reply bindings/rust/evmc-vm/src/lib.rs]: Created issue #308.

**Pull request merged**